src/sentry/incidents/logic.py

```diff
@@ -741,15 +770,23 @@ def update_alert_rule(
         updated_fields["team_id"] = owner.team_id if owner else None
         updated_fields["user_id"] = owner.user_id if owner else None
     if comparison_delta is not NOT_SET:
-        resolution = DEFAULT_ALERT_RULE_RESOLUTION
+        window: int = time_window or int(alert_rule.snuba_query.time_window / 60)
+        resolution = get_alert_resolution(window, organization=alert_rule.organization)
         if comparison_delta is not None:
             # Since comparison alerts make twice as many queries, run the queries less frequently.
-            resolution = DEFAULT_CMP_ALERT_RULE_RESOLUTION
+            resolution = resolution * DEFAULT_CMP_ALERT_RULE_RESOLUTION_MULTIPLIER
             comparison_delta = int(timedelta(minutes=comparison_delta).total_seconds())
 
         updated_query_fields["resolution"] = timedelta(minutes=resolution)
         updated_fields["comparison_delta"] = comparison_delta
 
+    if time_window and comparison_delta is NOT_SET and alert_rule.comparison_delta is not None:
```

Comment:
Can we add a comment here explaining this?

"If we're updating the time_window, and the comparison_delta, AND there's an existing comparison delta we're overriding.... then update our resolution"
?

Reply:
:squint:

wait. Can we just move this `and alert_rule.comparison_delta is not None` into the previous if statement?
so we only need to set the resolution once?

Reply:
nah, i'll add a comment (almost did before i opened the review too, thanks for confirming it's needed!)

---

Basically what's happening here is we're updating an alerts time_window. That would mean we need to also update the resolution field, which is happening on line 745. However, it's not taking into account if there is / isn't a comparison at that point (since there's a comparison handler later).

That comparison handler only checks if the comparison has _updated_.

This statement is saying "If we just updated the resolution because of a change to the time window, make sure that it wasn't previously a comparison query. if it was, use the multiplier on the resolution"

Reply:
oh duh
`is NOT_SET` vs. `is not NOT_SET`
totally missed the lowercase `not` lol

Reply:
@nhsiehgit yooo right? the read ability of `not NOT_SET` is tricky.

---

@wedamija def love that refactor. thanks, i couldn't see the forrest through the trees on that one.